astropy/cosmology/funcs/comparison.py

```diff
@@ -0,0 +1,439 @@
+        format. A `str` is assumed to be the correct format to use when
+        converting.
+
+    Returns
+    -------
+    |Cosmology| or generator thereof
+
+    Raises
+    ------
+    TypeError
+        If 'cosmo' is not a |Cosmology| and 'format' equals `numpy.False_`.
+        If 'cosmo' is a |Cosmology| and 'format' is not `None` or equal to `numpy.True_`.
+    """
+    # Deal with private wrapper
+    if isinstance(cosmo, _CosmologyWrapper):
+        cosmo = cosmo.wrapped
+
+    # Shortcut if already a cosmology
+    if isinstance(cosmo, Cosmology):
+        if format not in _COSMO_AOK:
+            allowed = '/'.join(map(str, _COSMO_AOK))
+            raise ValueError(f"for a Cosmology, 'format' must be {allowed}, not {format}")
+        return cosmo
+
+    if format != False_:  # catches False and False_
+        format = None if format == True_ else format  # str->str, None/True/True_->None
+        out = Cosmology.from_format(cosmo, format=format)  # this can error!
+    elif not isinstance(cosmo, Cosmology):
+        raise TypeError(f"if 'format' is False, arguments must be a Cosmology, not {cosmo}")
+    else:
+        out = cosmo
+
+    return out
+
+
+def _parse_formats(*cosmos: object, format: _FormatsT) -> ndarray:
+    """Parse Cosmology-like to |Cosmology|, using provided formats.
+
+    ``format`` is broadcast to match the shape of the cosmology arguments. Note
+    that the cosmology arguments are not broadcast against ``format``, so it
+    cannot determine the output shape.
+
+    Parameters
+    ----------
+    *cosmos : |Cosmology|-like
+        The objects to compare. Must be convertible to |Cosmology|, as specified
+        by the corresponding ``format``.
+
+    format : bool or None or str or array-like thereof, positional-only
+        Whether to allow, before equivalence is checked, the object to be
+        converted to a |Cosmology|. This allows, e.g. a |Table| to be equivalent
+        to a Cosmology. `False` (default) will not allow conversion. `True` or
+        `None` will, and will use the auto-identification to try to infer the
+        correct format. A `str` is assumed to be the correct format to use when
+        converting. Note 'format' is broadcast as an object array to match the
+        shape of 'cosmos' so 'format' cannot determine the output shape.
+
+    Raises
+    ------
+    TypeError
+        If any in 'cosmos' is not a |Cosmology| and the corresponding 'format'
+        equals `numpy.False_`.
+    """
+    formats = np.broadcast_to(np.array(format, dtype=object), len(cosmos))
+    # parse each cosmo & format
+
+    # Have to deal with things that do not broadcast well.
+    # astropy.row cannot be used in an array, even if dtype=object
+    # and will raise a segfault when used in a ufunc.
+    towrap = (isinstance(cosmo, _CosmologyWrapper._cantbroadcast) for cosmo in cosmos)
+    wcosmos = [c if not wrap else _CosmologyWrapper(c) for c, wrap in zip(cosmos, towrap)]
```

Comment:
It is possible to get rid of `not` by reordering:
```suggestion
    wcosmos = [_CosmologyWrapper(c) if wrap else c for c, wrap in zip(cosmos, towrap)]
```

Reply:
In this case I think I prefer the original order. I try to have minimal logic for clear flow, but in list comprehension I like to have the most common option first. Except for the few cases that need to be wrapped, this is ``[c for c in cosmos]``.